js/core.js

```diff
@@ -642,7 +642,9 @@ var ERMrest = (function (module) {
             var uri = this._getBaseURI(api);
 
             if (filter !== undefined && filter !== null) {
-                uri = uri + "/" + filter.toUri();
+                var filterUri = filter.toUri();
+                // Set filter in url only if the filterURI is not empty
+                if (filterUri.trim().length) uri = uri + "/" + filterUri
```

Comment:
If you apply trim to a string and it is reduced to `""`, it will still return a length of 0. Try having it check for greater than that or `!= -1`.